packages/web-runtime/tests/components/Notification.spec.js

```diff
@@ -0,0 +1,235 @@
+      expect(messageElement.text()).toBe('This is a test message')
+    })
+    it('displays the notification subject', () => {
+      const subjectElement = wrapper.find(selectors.subject)
+      expect(subjectElement.exists()).toBeTruthy()
+      expect(subjectElement.text()).toBeTruthy()
+    })
+  })
+
+  describe('active notifications actions', () => {
+    describe('when active notification action is empty', () => {
+      let wrapper, store
+      beforeEach(() => {
+        store = getStoreWithEmptyActions()
+        const mocks = { $store: store }
+        wrapper = mount(Notifications, {
+          localVue,
+          mocks
+        })
+        store.reset()
+      })
+      it('does not displays action button', () => {
+        expect(wrapper.find(selectors.actionButton).exists()).toBeFalsy()
+      })
+      it('displays the resolve notification button', () => {
+        expect(wrapper.find(selectors.resolveNotificationButton).exists()).toBeTruthy()
+      })
+      it('dispatches deleteNotification when resolve notification is clicked', () => {
+        const resolveNotificationButton = wrapper.find(selectors.resolveNotificationButton)
+        expect(resolveNotificationButton.exists()).toBeTruthy()
+        resolveNotificationButton.trigger('click')
+        wrapper.vm.$nextTick(() => {
+          expect(store.dispatch).toHaveBeenCalledTimes(1)
+          expect(store.dispatch).toHaveBeenCalledWith('deleteNotification', {
+            client: localVue.prototype.$client,
+            notification: 1
+          })
+        })
+      })
+    })
+    describe('when active notification action has a action', () => {
+      let store, wrapper
+      const mockMethods = {
+        executeRequest: jest.fn(),
+        reloadFilesList: jest.fn()
+      }
+      beforeEach(() => {
+        store = getStoreWithSingleAction()
+        store.reset()
+        const mocks = { $store: store }
+        wrapper = mount(Notifications, {
+          localVue,
+          methods: mockMethods,
+          mocks
+        })
+      })
+      it('displays the action button with action label', () => {
+        const actionButton = wrapper.find(selectors.actionButton)
+        expect(actionButton.exists()).toBeTruthy()
+        expect(actionButton.text()).toBe('Test Action')
+      })
+      it("doesn't displays the resolve notification button", () => {
+        expect(wrapper.find(selectors.resolveNotificationButton).exists()).toBeFalsy()
+      })
+      it('calls executeRequest when action button is clicked', () => {
+        const actionButton = wrapper.find(selectors.actionButton)
+        actionButton.trigger('click')
+        actionButton.vm.$nextTick(() => {
+          expect(mockMethods.executeRequest).toHaveBeenCalledTimes(1)
+        })
+      })
+    })
+  })
+  describe('when active notification action has a link', () => {
+    let store
+    afterEach(() => {
+      store.reset()
+    })
+    it('does not display link if no link is given', () => {
```

Comment:
ìt's inside `when active notification action has a link` why is here again no link given?